recipe-hooks-project/src/PrivateRoutes.js

```diff
@@ -0,0 +1,41 @@
+import React, {useContext} from "react";
+import {Redirect, Route} from "react-router-dom";
+import AppPage from "./AppPage";
+import Home from "./Home";
+import Register from "./Register";
+import Profile from "./Profile";
+import Ingredients from "./Ingredients";
+import Tags from "./Tags";
+import Recipes from "./Recipes";
+import RecipeDetails from "./RecipeDetails";
+import {AuthContext} from "./contexts/AuthenticationProvider";
+
+export function PrivateRoutes() {
+  const {token} = useContext(AuthContext);
+
+  return (
+    <Route
+      render={() =>
+        token !== null
```

Comment:
Probably `token !== null` is a bit too strict, could just be `token ? <>{/* private */}</> : <>{/* public */}</>`